### PR TITLE
Replace yarn with pnpm

### PR DESCRIPTION
**Pull request by arieljbirnbaum**
### Context

Needed to unblock #2101
Implements P-141

### Notable Changes

- `corepack enable pnpm` is used to add `pnpm` to the `PATH` so it can be invoked directly (i.e. without `corepack pnpm`).
- Looks like `only-allow pnpm` does _not_ prevent `npm install` from running :( `yarn install` OTOH is already blocked by the `packageManager` prop. Hence, dropping the `preinstall` hook.
- `pnpm run` doesn't automatically find scripts in workspace (i.e. like `yarn run` does if the script name includes a colon); we need to pass the workspace explicitly (with the `--filter` option)

### TODO

- [ ] Regenerate API defs (if needed)

### CODE REVIEW

File: scripts/ts-utils/move-vcregistry-snapshot.ts
Comment (arieljbirnbaum) [inline scripts/ts-utils/move-vcregistry-snapshot.ts:1]: Just updated the comment on top; prettier took over after that :D

**Review by Kailai-Wang — APPROVED** [review]
I'm fine with it as long as CI is happy

Feel free to resolve the conflict and merge

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -475,6 +478,8 @@ jobs:
         uses: docker/build-push-action@v5
         with:
           context: .
+          build-contexts: local-stash=docker-image://localhost:5000/local-stash:latest
+          load: true
```
Comment (Kailai-Wang) [inline .github/workflows/ci.yml:482]: We don't need it anymore - but let's delete it in the next PR :)

Comment (0xverin) [reply .github/workflows/ci.yml:482]: same as line 492&&493

**Pull request merged**